### PR TITLE
Example of adding custom headers to reporting client

### PR DESCRIPTION
**Pull request by danielamiao**

### CODE REVIEW

File: common/src/main/java/com/lightstep/tracer/shared/Options.java [hunk 1]
```diff
@@ -214,6 +219,15 @@ public OptionsBuilder withTag(String key, Object value) {
         }
 
         /**
+         * Sets user-defined key-value pairs that should be sent in the HTTP header of all of the
+         * reports produced by this tracer.
+         */
+        public OptionsBuilder withCustomHeader(String key, String value) {
+            customHeaders.put(key, value);
+            return this;
+        }
```
Comment (Hesamedin) [inline common/src/main/java/com/lightstep/tracer/shared/Options.java]: We might have anothre helper method that accepts `Map<String, String>` that lets user to create a map himselfe and inject it here.

**Comment by carlosalberto** [comment]
Lets have this rounded up and merged. @malafeev think you can help shepherd this to be up-to-date?

**Comment by malafeev** [comment]
sure

**Comment by malafeev** [comment]
@carlosalberto could you grant me write access so I can update branch?

**Comment by malafeev** [comment]
moved PR to java-common: https://github.com/lightstep/lightstep-tracer-java-common/pull/147
because all modified classes now in java-common

**Pull request closed without merging**